pydantic/_internal/_model_construction.py

```diff
@@ -361,26 +361,8 @@ def __fields__(self) -> dict[str, FieldInfo]:
             PydanticDeprecatedSince20,
             stacklevel=2,
         )
-        return self.model_fields
-
-    @property
-    def model_fields(self) -> dict[str, FieldInfo]:
-        """Get metadata about the fields defined on the model.
-
-        Returns:
-            A mapping of field names to [`FieldInfo`][pydantic.fields.FieldInfo] objects.
-        """
         return getattr(self, '__pydantic_fields__', {})
```

Comment:
We followed the pattern of `__fields__` above, can we follow the same pattern for that as well (removal, and deprecation in `main.py`)?

Reply:
Hum, not sure what you mean. Moving `__fields__` on `BaseModel`, and use the `deprecated_instance_property` decorator as well?

Reply:
I think it won't be possible because we want `__fields__` to have the following deprecation message:

> The `__fields__` attribute is deprecated, use `model_fields` instead

And using the decorator will "override" this message with

> Accessing this attribute on the instance is deprecated, and will be removed in Pydantic V3.

Reply:
That's fine, we can just deprecate fully in v3 then.